vaadin-dialog-flow-parent/vaadin-dialog-flow/src/main/java/com/vaadin/flow/component/dialog/Dialog.java

```diff
@@ -995,4 +995,36 @@ public Style getStyle() {
                 "Dialog does not support adding styles to overlay");
     }
 
+    /**
+     * Close all the Dialog components associated within the current UI.
+     */
+    public static void closeAllDialogs() {
+        if (UI.getCurrent() == null || !UI.getCurrent().isAttached()) {
+            throw new IllegalStateException(
+                    "Can't remove Dialogs when current UI is null or detached.");
+        }
+        UI.getCurrent().getChildren().forEach(component -> {
+            if (component instanceof Dialog) {
+                Dialog dialog = (Dialog) component;
+                if (dialog.isModal()) {
+                    removeModals(dialog);
+                }
+                dialog.close();
+            }
+        });
+    }
+
+    private static void removeModals(Component modalComponent) {
+        if (modalComponent instanceof HasComponents) {
+            modalComponent.getChildren().forEach(component -> {
+                if (component instanceof Dialog) {
```

Comment:
Minor thing. This could be refactored to an additional private method, as it shares the same logic with the one above. It would allow to use a method reference as well.

```java

    public static void closeAllDialogs() {
        if (UI.getCurrent() == null || !UI.getCurrent().isAttached()) {
            throw new IllegalStateException(
                    "Can't remove Dialogs when current UI is null or detached.");
        }
        UI.getCurrent().getChildren().forEach(Dialog::closeDialog);
    }

    private static void removeModals(Component modalComponent) {
        if (modalComponent instanceof HasComponents) {
            modalComponent.getChildren().forEach(Dialog::closeDialog);
        }
    }

    private static void closeDialog(Component component) {
        if (component instanceof Dialog) {
            Dialog dialog = (Dialog) component;
            if (dialog.isModal()) {
                removeModals(dialog);
            }
            dialog.close();
        }
    }
```